pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -67,7 +69,19 @@ func (e ShortcutExpander) getAll() []schema.GroupResource {
 		}
 	}
 
-	return availableAll
+	// populate all "discovered" resources from the server
+	for availableResource := range availableResources {
```

Comment:
not sure defaulting to all discovered resources is correct, given the [current definition of "all"](https://github.com/kubernetes/community/blob/master/contributors/devel/kubectl-conventions.md#rules-for-extending-special-resource-alias---all)